packages/eupgrade/files/usr/lib/lua/eupgrade.lua

```diff
@@ -0,0 +1,177 @@
+local utils = require "lime.utils"
+local json = require "luci.jsonc"
+local libuci = require "uci"
+
+local eup = {}
+
+eup.STATUS_DEFAULT = 'not-initiated'
+eup.STATUS_DOWNLOADING = 'downloading'
+eup.STATUS_DOWNLOADED = 'downloaded'
+eup.STATUS_DOWNLOAD_FAILED = 'download-failed'
+
+
+local uci = libuci.cursor()
+
+function eup.set_workdir(workdir)
+    if not utils.file_exists(workdir) then
```

Comment:
what if exists but it is not dir?

Reply:
What do you suggest, to loga warning if it is not a directory?

Reply:
in that case it probably should fail all together and report an error